packages/hub/src/lib/LazyBlob.ts

```diff
@@ -0,0 +1,92 @@
+import { createReadStream } from "node:fs";
+import { open } from "node:fs/promises";
+import type { FileHandle } from "node:fs/promises";
+import { Readable } from "node:stream";
+
+export class LazyBlob extends Blob {
+	static async create(path: string, start?: number, end?: number): Promise<LazyBlob> {
+		const lazyBlob = new LazyBlob(path, start, end);
+
+		await lazyBlob.init();
+
+		return lazyBlob;
+	}
+
+	private path: string;
+	private start: number | null;
+	private end: number | null;
+	private totalSize: number;
+
+	private constructor(path: string, start?: number, end?: number) {
+		super();
+
+		this.path = path;
+		this.start = start || null;
+		this.end = end || null;
+
+		this.totalSize = 0;
+	}
+
+	get size(): number {
+		if (this.start !== null) {
+			if (this.end !== null) {
+				return Math.abs(this.end - this.start);
+			}
+
+			return this.totalSize - this.start;
+		}
+
+		return this.totalSize;
+	}
+
+	get type(): string {
+		return "";
+	}
+
+	slice(start = 0, end = this.size): LazyBlob {
+		const slice = new LazyBlob(this.path, start, end);
+
+		return slice;
+	}
+
+	async blob(): Promise<Blob> {
+		return this.slice(0, this.size);
+	}
+
+	async arrayBuffer(): Promise<ArrayBuffer> {
+		const start = this.start || 0;
+		const end = this.end || this.size;
+		const size = Math.abs(end - start);
+		const cappedSize = size > this.size ? this.size : size;
+
+		const slice = await this.execute((file) => file.read(Buffer.alloc(cappedSize), 0, cappedSize, 0));
```

Comment:
```suggestion
		const slice = await this.execute((file) => file.read(Buffer.alloc(this.size), 0, this.size, this.start));
```

With the other requested change on the blob creation, it should be possible to simplify the code to this